src/com/amazon/kinesis/streaming/agent/providers/CLIWebIdentityTokenCredentialsProvider.java

```diff
@@ -0,0 +1,108 @@
+package com.amazon.kinesis.streaming.agent.providers;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.joda.time.format.DateTimeFormatter;
+import org.joda.time.format.DateTimeFormatterBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.BasicSessionCredentials;
+
+/**
+ * This provider uses the AWS CLI to call assume-role-with-web-identity to retrieve credentials.
+ * Refresh on expiration is implemented.
+ */
+public class CLIWebIdentityTokenCredentialsProvider implements AWSCredentialsProvider {
```

Comment:
don't you need to add this to the credsProviderChain?

Reply:
I was hesitant to change existing functionality. I figured whoever needs to use it can add it simply by specifying the `userDefinedCredentialsProvider.classname` configuration setting for the agent.